### PR TITLE
RFC: Split Python API doc in subpages

### PR DESCRIPTION
**Pull request by PicoCentauri**
Since a single operations for the Python operations becomes quite big this PR splits this single pages into different sub pages. Additionally, also the `block` and the `miscellaneous` page is also split.

With this change each function also becomes visible and in the menu.

If we agree on this new structure I will adapt this also for the C/C++ references.

### CODE REVIEW

File: docs/src/reference/python/data.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Data arrays
+===========
+
+.. autoclass:: equistore.data.Array
+    :members:
```
Comment (Luthaf) [inline docs/src/reference/python/data.rst:5]: for some reason this renders as empty? I though we had docs for this, even if they are minimal

File: docs/src/reference/python/operations/index.rst [hunk 2]
```diff
@@ -0,0 +1,12 @@
+Operations
+==========
+
+.. automodule:: equistore.operations
+    :undoc-members:
+
+.. toctree::
```
Comment (Luthaf) [inline docs/src/reference/python/operations/index.rst:7]: Can we have a `:maxdepth: 1` here?

Comment (PicoCentauri) [reply docs/src/reference/python/operations/index.rst:7]: Yes.

File: docs/src/reference/python/data.rst (hunk 1 above)
Comment (PicoCentauri) [reply docs/src/reference/python/data.rst]: Weird. It renders locally fine. I changed it to exactly the same line as before the PR. Lets see what RTD is doing now...

Comment (PicoCentauri) [reply docs/src/reference/python/data.rst]: Hmm still seems to render properly. A reason might be that the docs are only injected if `HAS_TORCH=True` in `python/src/equistore/data/extract.py`. But why did it worked before...

**Pull request merged**